src/daft-sql/src/planner.rs

```diff
@@ -1860,23 +1865,23 @@ impl<'a> SQLPlanner<'a> {
             // ---------------------------------
             // signed integer
             // ---------------------------------
-            SQLDataType::Int(_) | SQLDataType::Integer(_) => DataType::Int32,
-            SQLDataType::Int2(_) | SQLDataType::SmallInt(_) => DataType::Int16,
-            SQLDataType::Int4(_) | SQLDataType::MediumInt(_) => DataType::Int32,
-            SQLDataType::Int8(_) | SQLDataType::BigInt(_) => DataType::Int64,
             SQLDataType::TinyInt(_) => DataType::Int8,
+            SQLDataType::Int2(_) | SQLDataType::SmallInt(_) | SQLDataType::Int16 => DataType::Int16,
+            SQLDataType::Int(_) | SQLDataType::Integer(_) | SQLDataType::Int4(_) | SQLDataType::MediumInt(_)  | SQLDataType::Int32  => DataType::Int32,
+            SQLDataType::Int8(_) | SQLDataType::BigInt(_) | SQLDataType::Int64 => DataType::Int64,
```

Comment:
Since this is adding bit-precision aliases, what do you think of not including the byte-precision aliases? To me having `int8 > int16` is hazardous for customers. Neither are SQL standard, and I would prefer bit-precision aliases because of arrow. We could give nice error messages for these cases such as a `INT8 -> Did you mean INT64?`.